### PR TITLE
Limit CPU utilization of jenkins tests.

### PR DESCRIPTION
**Pull request by robertnishihara**
This attempts to address #2595. cc @shaneknapp 

Note that passing `--cpus=12` into `docker run` is not *exactly* the behavior that we want. I think it will limit the docker containers CPU utilization, but `nproc` will still return the full number of CPUs so Ray will still try to create 48 workers (or however many cores there are).

### CODE REVIEW

**Comment by shaneknapp** [comment]
i took a look at the stats on worker-02, and things are definitely looking better (~35 system load).

i also understand nproc overriding things, but we should still investigate how to limit how much ray takes over a system when it's running on it.  :)

File: test/jenkins_tests/run_multi_node_tests.sh [hunk 1]
```diff
@@ -11,283 +11,284 @@ ROOT_DIR=$(cd "$(dirname "${BASH_SOURCE:-$0}")"; pwd)
 DOCKER_SHA=$($ROOT_DIR/../../build-docker.sh --output-sha --no-cache)
 echo "Using Docker image" $DOCKER_SHA
 
-docker run  -e "RAY_USE_XRAY=1" --rm --shm-size=10G --memory=10G $DOCKER_SHA \
+docker run  -e "RAY_USE_XRAY=1" --rm --cpus=12 --shm-size=10G --memory=10G $DOCKER_SHA \
     python /ray/python/ray/rllib/train.py \
     --env PongDeterministic-v0 \
     --run A3C \
     --stop '{"training_iteration": 2}' \
     --config '{"num_workers": 16}'
```
Comment (ericl) [inline test/jenkins_tests/run_multi_node_tests.sh]: Note that setting num CPUs won't don't anything, you have to set num workers. Here it seems to be 16, but could be changed to 2.

Comment (robertnishihara) [reply test/jenkins_tests/run_multi_node_tests.sh]: I think what it should do is limit the total CPU utilization of the docker container (Ray will still create more workers). Does that sound right?

Comment (robertnishihara) [reply test/jenkins_tests/run_multi_node_tests.sh]: I haven't actually verified that.

**Comment by ericl** [comment]
Yes, but you could also just change the num workers setting to be lower
there. There's no good reason for the tests to use more than a couple CPU
anyways.

**Comment by richardliaw** [comment]
I think it's probably reasonable to do both num_workers and CPU; there are
some examples that don't limit the threading for the TF models, resulting
in 1 worker using >100% CPU utilization.

I'm fairly certain the CPUs arg for docker _will_ limit the container
utilization (see this thread
<https://stackoverflow.com/questions/51677678/any-tool-to-parallelize-bash-commands-with-different-cpu-requirements/51677777#51677777>).
In fact, for stress tests (not RLlib or Tune tests), we should consider
using high number of workers and small CPU.

Another thing that is probably worth considering is limiting the
OMP_NUM_THREADS variable, as seen in previous issues.

**Comment by shaneknapp** [comment]
from `man nproc`:

```       --ignore=N
              if possible, exclude N processing units
```

so maybe we can use this to limit how many available cores to use?

```sknapp@amp-jenkins-staging-worker-02:~$ nproc --ignore=40
8```

i'm seeing this called in the root `build.sh` script, so maybe we can hack that to accept CLI args to accept core limiting?

**Review by ericl — APPROVED** [review]

**Comment by robertnishihara** [comment]
jenkins, retest this please

**Comment by robertnishihara** [comment]
Thanks @shaneknapp I messed up the rebase. I want to double check that these tests are passing and that I didn't introduce a bug.

**Comment by shaneknapp** [comment]
test this please

**Comment by richardliaw** [comment]
retest this please

**Comment by richardliaw** [comment]
jenkins retest this please

**Comment by robertnishihara** [comment]
I should probably just revert the change to `many_drivers_test.py`, since it seems to be hanging.

**Comment by richardliaw** [comment]
retest this please

**Comment by richardliaw** [comment]
@robertnishihara are you planning to merge this?

**Comment by robertnishihara** [comment]
Closing due to staleness. If this is still an issue we can still do this fix.

**Pull request closed without merging**